runtime/src/tasks/qthreads/tasks-qthreads.c

```diff
@@ -732,33 +731,70 @@ static inline void wrap_callbacks(chpl_task_cb_event_kind_t event_kind,
     }
 }
 
-static aligned_t chapel_wrapper(void *arg)
+// If we stored chpl_taskID_t in chpl_task_bundleData_t,
+// this struct and the following function may not be necessary.
+typedef void (*main_ptr_t)(void);
+typedef struct {
+  chpl_task_bundle_t arg;
```

Comment:
The argument doesn't actually have to follow the fixed fields, that was something I was predicting we would need but didn't end up needing.
It turns out that all of our on/task bodies know the argument bundle type they are starting with. So in this case that means that main_wrapper can just cast its argument to main_wrapper_bundle_t*. We need chpl_task_bundle_t to be the first structure member though so that the runtime can access those fields.